aderyn_core/src/detect/low/function_pointer_in_constructor.rs

```diff
@@ -99,6 +99,22 @@ mod func_compilation_solc_pragma_helper {
             }
             false
         }
+        pub fn compiles_for_solc_below_0_6_5(&self, context: &WorkspaceContext) -> bool {
+            if let Some(source_unit) = self.closest_ancestor_of_type(context, NodeType::SourceUnit)
+            {
+                let pragma_directives = ExtractPragmaDirectives::from(source_unit).extracted;
+
+                if let Some(pragma_directive) = pragma_directives.first() {
```

Comment:
We'll bring these helper functions into a helpers module in a separate PR so it's visible